sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -566,6 +566,39 @@ val GPU_COREDUMP_PIPE_PATTERN = conf("spark.rapids.gpu.coreDump.pipePattern")
       s"Batch size must be positive and not exceed ${Integer.MAX_VALUE} bytes.")
     .createWithDefault(1 * 1024 * 1024 * 1024) // 1 GiB is the default
 
+  val CHUNKED_READER = conf("spark.rapids.sql.reader.chunked")
+    .doc("Enable a chunked reader where possible. A chunked reader allows " +
+      "reading highly compressed data that could not be read otherwise, but at the expense " +
+      "of more GPU memory, and in some cases more GPU computation. "+
+      "Currently this only supports ORC and Parquet formats.")
+    .booleanConf
+    .createWithDefault(true)
+
+  val CHUNKED_READER_MEMORY_USAGE_RATIO = conf("spark.rapids.sql.reader.chunked.memoryUsageRatio")
+    .doc("A value to compute soft limit on the internal memory usage of the chunked reader " +
+      "(if being used). Such limit is calculated as the multiplication of this value and " +
+      s"'${GPU_BATCH_SIZE_BYTES.key}'.")
+    .doubleConf
+    .checkValue(v => v > 0, "The ratio value must be positive.")
+    .createWithDefault(4)
+
+  val LIMIT_CHUNKED_READER_MEMORY_USAGE = conf("spark.rapids.sql.reader.chunked.limitMemoryUsage")
+    .doc("Enable a soft limit on the internal memory usage of the chunked reader " +
+      "(if being used). Such limit is calculated as the multiplication of " +
+      s"'${GPU_BATCH_SIZE_BYTES.key}' and '${CHUNKED_READER_MEMORY_USAGE_RATIO.key}'." +
+      "For example, if batchSizeBytes is set to 1GB and memoryUsageRatio is 4, " +
+      "the chunked reader will try to keep its memory usage under 4GB.")
+    .internal()
+    .booleanConf
+    .createOptional
```

Comment:
this should be marked as `startupOnly` as it's used within a lazy val here: https://github.com/NVIDIA/spark-rapids/pull/10718/files#diff-12777fe3bbb26d1a892f120acf176b07af222570dafd19f039adc3f84d4c58e2R2554

Reply:
Humnn, it seems that removing `lazy` is not enough but these variables should be changed into function to re-evaluate the config every time.

Reply:
Just know that a new instance of `RapidsConf` will be created if the configs change, thus static context doesn't matter here.

Reply:
yeah never mind! Sorry for the trouble.